codec-http2/src/main/java/io/netty/handler/codec/http2/UniformStreamByteDistributor.java

```diff
@@ -97,10 +97,13 @@ public boolean distribute(int maxBytes, Writer writer) throws Http2Exception {
         State state = queue.pollFirst();
         do {
             state.enqueued = false;
-            if (state.streamableBytes > 0 && maxBytes == 0) {
-                // Stop at the first state that can't send. Add this state back to the head of
-                // the queue. Note that empty frames at the head of the queue will always be
-                // written.
+            if (!state.writeAllowed) {
```

Comment:
Maybe instead of using another variable `writeAllowed`, it might be more clear if we just use `windowSize` directly:

```java
if (state.windowSize <= 0) {
    continue;
}
```